onnxruntime/core/providers/cpu/controlflow/loop.h

```diff
@@ -26,9 +26,20 @@ class Loop final : public OpKernel, public controlflow::IControlFlowKernel {
   struct Info;
   ~Loop();
 
+  // function to concatenate the OrtValue instances from each Loop iteration into a single output buffer.
+  // @param per_iteration_output OrtValue instances from each iteration. Never empty. All should have the same shape.
```

Comment:
Just curious: Is it valid to have 0 `scan_outputs` (and hence empty `per_iteration_output`) ?

Reply:
it's possible for Loop to have zero 'scan outputs'. we only call the concatenate method for actual scan outputs. it can also have zero iterations, which would lead to having empty per iteration output, but that is special cased and handled separately (the output will have a value of 0 in the first dim so there's no data to do anything with - just a shape to define)